apps/framework-cli/src/infrastructure/olap/clickhouse.rs

```diff
@@ -901,7 +901,7 @@ fn convert_clickhouse_type_to_column_type(ch_type: &str) -> Result<(ColumnType,
     }
 
     // Handle DateTime types with parameters
-    if ch_type.starts_with("DateTime") {
+    if ch_type.starts_with("DateTime") || ch_type == "Date32" {
```

Comment:
I don't like that this function skips over `ClickHouseColumnType::from_type_str` and have duplicate logic.

Reply:
ie you want to refactor in a follow up?

Reply:
yap